modules/im_cloudbuild_workspace/github.tf

```diff
@@ -24,7 +24,7 @@ locals {
   existing_github_secret_version = local.is_gh_repo && var.github_pat_secret != "" ? data.google_secret_manager_secret_version.existing_github_pat_secret_version[0].name : ""
   github_secret_version_id       = local.create_github_secret ? google_secret_manager_secret_version.github_token_secret_version[0].name : local.existing_github_secret_version
 
-  secret_id = var.github_personal_access_token != "" ? google_secret_manager_secret.github_token_secret[0].id : data.google_secret_manager_secret.existing_github_pat_secret[0].secret_id
+  github_secret_id = local.is_gh_repo ? (local.create_github_secret ? google_secret_manager_secret.github_token_secret[0].id : data.google_secret_manager_secret.existing_github_pat_secret[0].secret_id) : ""
```

Comment:
nit: `alltrue` might simplify logic here alltrue([local.is_gh_repo, local.create_github_secret]) ? ...

Reply:
I do need the `local.is_gh_repo` check to happen immediately for setting this to an empty string, otherwise there'll be Terraform errors for trying to access resources that may not exist (if using GitLab instead).